### PR TITLE
Validate forms using DTO for access, asset profile, tag, and platform management

### LINKED ISSUE #3294: Adopt validation with DTO for various forms
Adopt concept of form validation with DTO for the access, asset profile, tag and platform management.

_Originally posted by @dtslvr in https://github.com/ghostfolio/ghostfolio/pull/3230#pullrequestreview-1999699948_

### PR DESCRIPTION
**Pull request by Fedron**
Resolves #3294

Validate forms using DTO for access, asset profile, tag, and platform management

### CODE REVIEW

File: apps/client/src/app/components/admin-platform/admin-platform.component.ts
Comment (dtslvr) [inline apps/client/src/app/components/admin-platform/admin-platform.component.ts:1]: 👍🏻

File: apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts [hunk 1]
```diff
@@ -11,13 +21,48 @@ import { CreateOrUpdatePlatformDialogParams } from './interfaces/interfaces';
   styleUrls: ['./create-or-update-platform-dialog.scss'],
   templateUrl: 'create-or-update-platform-dialog.html'
 })
-export class CreateOrUpdatePlatformDialog {
+export class CreateOrUpdatePlatformDialog implements OnDestroy {
+  public platformForm: FormGroup;
   private unsubscribeSubject = new Subject<void>();
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts:26]: Add a line break please

File: apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts [hunk 2]
```diff
@@ -11,13 +21,48 @@ import { CreateOrUpdatePlatformDialogParams } from './interfaces/interfaces';
...
+    public dialogRef: MatDialogRef<CreateOrUpdatePlatformDialog>,
+    private formBuilder: FormBuilder
+  ) {
+    this.platformForm = this.formBuilder.group({
+      name: [this.data.platform.name, Validators.required],
+      url: [this.data.platform.url, Validators.required]
+    });
+  }
+
+  public async onSubmit() {
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts:39]: Can you move below `onCancel()`?

File: apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts [hunk 3]
```diff
@@ -11,13 +21,46 @@ import { CreateOrUpdateTagDialogParams } from './interfaces/interfaces';
   styleUrls: ['./create-or-update-tag-dialog.scss'],
   templateUrl: 'create-or-update-tag-dialog.html'
 })
-export class CreateOrUpdateTagDialog {
+export class CreateOrUpdateTagDialog implements OnDestroy {
+  public tagForm: FormGroup;
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts:25]: Add a line break please

File: apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts [hunk 4]
```diff
@@ -11,13 +21,46 @@ import { CreateOrUpdateTagDialogParams } from './interfaces/interfaces';
...
-  ) {}
+    public dialogRef: MatDialogRef<CreateOrUpdateTagDialog>,
+    private formBuilder: FormBuilder
+  ) {
+    this.tagForm = this.formBuilder.group({
+      name: [this.data.tag.name]
+    });
+  }
+
+  public async onSubmit() {
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts:38]: Can you move below `onCancel()`?

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts [hunk 5]
```diff
@@ -65,28 +66,38 @@ export class CreateOrUpdateAccessDialog implements OnDestroy {
...
+            if (error.status === StatusCodes.BAD_REQUEST) {
+              alert($localize`Oops! Could not grant access.`);
+            }
+
+            return EMPTY;
+          }),
+          takeUntil(this.unsubscribeSubject)
+        )
+        .subscribe(() => {
+          this.dialogRef.close({ access });
```
Comment (dtslvr) [inline apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts:96]: This is currently not consistent:
`this.dialogRef.close({ access });`
vs.
`this.dialogRef.close(platform);`

What do you prefer?

**Review by dtslvr — CHANGES_REQUESTED** [review]
Very nice! 😎

Can you adapt the [account](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/accounts/accounts-page.component.ts#L261) and [activities](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/portfolio/activities/activities-page.component.ts#L290) page as well to this more beautiful approach?

File: apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts (hunk 1 above)
Comment (Fedron) [reply apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts:27]: Added

File: apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts (hunk 2 above)
Comment (Fedron) [reply apps/client/src/app/components/admin-platform/create-or-update-platform-dialog/create-or-update-platform-dialog.component.ts:44]: Moved

File: apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts (hunk 3 above)
Comment (Fedron) [reply apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts:25]: Added

File: apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts (hunk 4 above)
Comment (Fedron) [reply apps/client/src/app/components/admin-tag/create-or-update-tag-dialog/create-or-update-tag-dialog.component.ts:43]: Moved

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts (hunk 5 above)
Comment (Fedron) [reply apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts]: To be honest I kind of prefer the second approach so I've updated the access dialog close to match.

**Review by dtslvr — APPROVED** [review]
Thanks again @fedron! I have made some minor improvements. Ready to merge now.

**Pull request merged**